internal/juju/jaas.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package juju
+
+import (
+	"github.com/canonical/jimm-go-sdk/v3/api"
+	"github.com/canonical/jimm-go-sdk/v3/api/params"
+	jujuapi "github.com/juju/juju/api"
+
+	"github.com/juju/errors"
+)
+
+type jaasClient struct {
+	SharedClient
+	getJaasApiClient func(jujuapi.Connection) JaasAPIClient
+}
+
+func newJaasClient(sc SharedClient) *jaasClient {
+	return &jaasClient{
+		SharedClient: sc,
+		getJaasApiClient: func(conn jujuapi.Connection) JaasAPIClient {
+			return api.NewClient(conn)
+		},
+	}
+}
+
+func (jc *jaasClient) AddRelations(tuples []params.RelationshipTuple) error {
+	conn, err := jc.GetConnection(nil)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = conn.Close() }()
```

Comment:
should we log the returned error?

Reply:
On close I think it's okay to ignore, it's also the pattern used elsewhere which I'm happy to keep.

Reply:
On conn.Close() it's okay to ignore the error. I prefer this to conn.Close() which causes compiler warnings.